sktime/forecasting/compose/_multiplexer.py

```diff
@@ -84,83 +84,114 @@ class MultiplexForecaster(_HeterogenousEnsembleForecaster):
         "handles-missing-data": False,
         "scitype:y": "both",
         "y_inner_mtype": ["pd.DataFrame", "pd.Series"],
+        "fit_is_empty": False,
     }
 
+    _delegate_name = "forecaster_"
+
     def __init__(
         self,
         forecasters: list,
         selected_forecaster=None,
     ):
-        super(MultiplexForecaster, self).__init__(forecasters=forecasters, n_jobs=None)
+        super(MultiplexForecaster, self).__init__()
         self.selected_forecaster = selected_forecaster
 
+        self.forecasters = forecasters
+        self.forecasters_ = self._check_estimators(
+            forecasters,
+            attr_name="forecasters",
+            cls_type=BaseForecaster,
+            clone_ests=False,
+        )
+        self._set_forecaster()
+        self.clone_tags(self.forecaster_)
+        self.set_tags(**{"fit_is_empty": False})
+
     def _check_selected_forecaster(self):
-        component_names = [name for name, _ in self.forecasters]
-        if self.selected_forecaster not in component_names:
+        component_names = self._get_estimator_names(self.forecasters_, make_unique=True)
+        selected = self.selected_forecaster
+        if selected is not None and selected not in component_names:
             raise Exception(
-                "Please check the selected_forecaster argument provided "
-                " Valid selected_forecaster parameters: {}".format(component_names)
+                f"Invalid selected_forecaster parameter value provided, "
+                f" found: {self.selected_forecaster}. Must be one of these"
+                f" valid selected_forecaster parameter values: {component_names}."
             )
 
     def _set_forecaster(self):
         self._check_selected_forecaster()
+        # clone the selected forecaster to self.forecaster_
         if self.selected_forecaster is not None:
-            for name, forecaster in self.forecasters:
+            for name, forecaster in self._get_estimator_tuples(self.forecasters):
                 if self.selected_forecaster == name:
                     self.forecaster_ = clone(forecaster)
+        else:
+            # if None, simply clone the first forecaster to self.forecaster_
```

Comment:
unfortunately, we cannot do that - according to the sklearn interface, one must not change hyper-parameters from what was passed. Hyper-parameters are considered, by design, immutable.

Reply:
Makes sense, as long as it is consistent/what users expect then I feel it should be fine!

Reply:
not necessarily a user expectation since most users are not familiar with the inner workings of sklearn - but it is documented, an consistent with sklearn